### ISSUE TITLE
Consider improving invalid JDBC URL error messages in class generation

### ISSUE THREAD

**Issue by perlun** [jOOQ/jOOQ#12944]
### Expected behavior
Invalid JDBC URLs encountered in class generation should print an error message to the user

### Actual behavior

```
> Task :myproject:generateJooqPojos FAILED
JDBC URL must contain a / at the end of the host or port: jdbc:postgresql://localhost
SQL exception            : Exception while executing meta query: Cannot execute query. No JDBC Connection configured

If you think this is a bug in jOOQ, please report it here: https://github.com/jOOQ/jOOQ/issues/new

Note you can mute some exceptions using the configuration/onError flag

```sql
select pg_catalog.pg_namespace.nspname
from pg_catalog.pg_namespace
order by pg_catalog.pg_namespace.nspname```
```

(I feel embarrassed now... When preparing this issue, I see that **it does actually print an error already**, the first line: `JDBC URL must contain a / at the end of the host or port: jdbc:postgresql://localhost`; I missed this completely until now. I think it was the fact that it then printed `Cannot execute query. No JDBC Connection configured` that got me confused. :see_no_evil: Perhaps we could make it clearer somehow, though? :thinking:)

I guess one issue here is that the overall output was so long here, both with the link to creating a bug report, hints about muting exceptions etc., that the part most essential to me (`JDBC URL must contain a / at the end of the host or port: jdbc:postgresql://localhost`) kind of "drowned" in all the output. I understand if it's impossible to special-case any and every potential problem here, but I see two ways to make the above better:

1. Silence all "noise" in the output above. I'm not sure if this is doable or not, perhaps by throwing a more specific exception (to avoid emitting all the "extra" output above) in cases where an invalid JDBC URL is detected.
2. If the above is considered to require too much effort, some more blank lines/special characters like `***` or `|||`, could help.

<sub>I can also live with this not being fixed at all. :slightly_smiling_face: I _did_ actually more or less deliberately give it (jooq) a slightly invalid JDBC URL. :grin: And it _did_ actually point me in the right direction, if I would just have read the output more carefully...</sub>

### Steps to reproduce the problem

I used the following Gradle build script to generate the classes:

```groovy
buildscript {
    repositories {
        mavenLocal()
        mavenCentral()
    }

    dependencies {
        classpath "org.jooq:jooq-codegen:3.16.2"
        classpath "org.postgresql:postgresql:42.3.1"
    }
}

import org.jooq.codegen.GenerationTool
import org.jooq.meta.jaxb.*

task generateJooqPojos() {
    doLast {
        GenerationTool.generate( new Configuration()
                .withJdbc( new Jdbc()
                        .withDriver( 'org.postgresql.Driver' )
                        .withUrl( 'jdbc:postgresql://localhost' ) // note the invalid JDBC URL here; missing slash at the end
                        .withUser( 'postgres' )
                        .withPassword( '<removed>' ) )
                .withGenerator( new Generator()
                        .withDatabase( new Database() )
                        .withGenerate( new Generate()
                                .withPojos( true )
                                .withDaos( true ) )
                        .withTarget( new Target()
                                .withPackageName( 'org.jooq.example.gradle.db' )
                                .withDirectory( '/tmp/java' ) ) ) )
```

### Versions

- jOOQ: 3.16.2
- Java: 11.0.13
- Database (include vendor): PostgreSQL 13 + TimescaleDB 2.5.0 (Docker image `timescale/timescaledb:2.5.0-pg13-oss`)
- OS: Debian GNU/Linux 11 (`bullseye`)
- JDBC Driver (include name if inofficial driver): `org.postgresql:postgresql:42.3.1`

**Comment by lukaseder** [comment]
Huh, tried this with a wrong JDBC URL and the `org.hsqldb.jdbcDriver`. It just returns a `null` connection, which we don't handle. I've created a separate issue for this: https://github.com/jOOQ/jOOQ/issues/12951

**Comment by lukaseder** [comment]
Oh, I see, it's actually the same issue as #12951. pgjdbc seems to think that a URL like `jdbc:postgresql://localhost` is not a pgjdbc URL, hence it returns a `null` connection from `Driver::connect` (which then produces the #12951 bug).

I think that's a bug in pgjdbc worth addressing there: https://github.com/pgjdbc/pgjdbc/issues/2421

In any case, with #12951 fixed, the problem would have been clearer more immediately.

**Comment by lukaseder** [comment]
#12951 is fixed in jOOQ 3.17.0, 3.16.4 (#12952), and 3.15.8 (#12953)

**Comment by perlun** [comment]
Thanks! :+1:
